### ISSUE TITLE
test if eclipse liked trip2dicom converted files

### ISSUE THREAD

**Issue by nbassler** [pytrip/pytrip#285]

**Comment by nbassler** [comment]
related: #266

**Comment by TokePrintz** [comment]
![errorinimport_pytrip134](https://cloud.githubusercontent.com/assets/22157203/23484341/d9a96dce-fed6-11e6-9ed3-e90dc74b51c7.png)
My first tests were unfortunately not very successful. Having tried with different files and both trip2dicom on dicom2trip-generated voxelplans as well as trip2dicom on legacy 1.2 format, I found that I cannot import the patients. I couldn't seem to find a workaround. And having never encountered this type of error before (basically at a stage in the import process where the patient has already been read and recognized), I am in new territory... 
I will have to look further into this tomorrow, maybe find a fresh new DICOM from the clinic and try to import that....just to figure out if that at least work...

**Comment by TokePrintz** [comment]
And with "pure" DICOM (ones that have not been generated by artificial means) I do not have any problems with import.
From the error it seems that the problem is that "ImageId" violates the maximum length of this tag. Is this something that has recently been changed? Because with version 1.3.2 it all worked fine.

**Comment by grzanka** [comment]
Interesting, in the example Eclipse data files from slicer project, ImageID tag (http://dicomlookup.com/lookup.asp?sw=Tnumber&q=(0054,0400)) is not set, we are also not setting it in pytrip

```
→ dcmdump ../data/eclipse-8.1.20-phantom-breast/Original/CT.1.2.840.113619.2.55.3.671756986.106.1316467036.362.1.dcm | grep Image    ~/PycharmProjects/pytrip
(0002,0002) UI =CTImageStorage                          #  26, 1 MediaStorageSOPClassUID
(0008,0008) CS [ORIGINAL\PRIMARY\AXIAL]                 #  22, 3 ImageType
(0008,0016) UI =CTImageStorage                          #  26, 1 SOPClassUID
(0020,0032) DS [-275\-275\107]                          #  14, 3 ImagePositionPatient
(0020,0037) DS [1\0.0\0.0\0.0\1\0.0]                    #  20, 6 ImageOrientationPatient
```

**Comment by grzanka** [comment]
In the Varian DICOM conformance document (https://www.varian.com/sites/default/files/resource_attachments/Eclipse_Ocular_Proton_Planning_11.0.pdf) I have found however this one:

![image](https://cloud.githubusercontent.com/assets/7374733/23527444/e6e908f2-ff8d-11e6-8e50-81834fa9107d.png)

In Eclipse example data its present:
```
→ dcmdump ../data/eclipse-8.1.20-phantom-breast/Original/RI.1.2.246.352.71.3.2088656855.2377794.20110920152340.dcm| grep RTImageLabel
(3002,0002) SH [MED BREL-DRR]                           #  12, 1 RTImageLabel
```

**Comment by grzanka** [comment]
Ups, this was for EOPP (Ocular), we should look here https://www.varian.com/sites/default/files/resource_attachments/Varian_System_Server_v13.0_DICOM_Conformance_Statement.pdf

**Comment by TokePrintz** [comment]
The test file that @nbassler  send my yesterday evening was working. I do not know what was changed however. 
Also, the Eclipse import has been working for all other late releases than 1.3.4 and as I understand it, in none of those was the ImageID tag set?

**Comment by grzanka** [comment]
What was changed in last releases its hard to trace without access to Eclipse.

Finally I went through main Varian conformance statement and found out that there no such tag as ImageID, but other tags play its role:

For RT images:
![image](https://cloud.githubusercontent.com/assets/7374733/23542985/f4cb9520-ffe7-11e6-93a0-e0f86593d122.png)

For nonRT images:
![image](https://cloud.githubusercontent.com/assets/7374733/23543019/1d666c6c-ffe8-11e6-96c0-80aeec2a98f2.png)


![image](https://cloud.githubusercontent.com/assets/7374733/23542999/04ccb9a4-ffe8-11e6-810d-d265d41a9046.png)

**Comment by TokePrintz** [comment]
Okay, first of all. The issue remains in 1.3.7.

We did dcmdump on the tags above on a working DICOM from 1.3.3 and one from 1.3.7, which cannot be imported. Tags (3002,0002) and (3002,0003) doesn't exist in either of them. (0020,0013) exists in both of them and have X instances of integers, with X being the number of slices (see image)
![screenshot](https://cloud.githubusercontent.com/assets/22157203/23543629/4e350352-fff3-11e6-9297-05356ddc6b77.png)
So I don't think this is the error. 

We will compare the 1.3.3 and 1.3.7 DICOMs to see if we can pinpoint the issue. 

We now know from Niels' test that the bug is related to the contours. It has to be a recent change in vdx.py. But going through the code I don't see anything suspicious.

**Comment by TokePrintz** [comment]
Okay, I found the error (in vdx.py as expected). It was very simple.
Before (1.3.3) we had "ds.StructureSetLabel = 'PyTRiP plan'"
In the RTSS fix it was changed to "ds.StructureSetLabel = 'PyTRiP Structures'"
Changing the StructureSetLabel from [PyTRiP Structures] back to [PyTRiP plan] using dcmmodify fixed the error. Now I can import the patient again. 
I am not sure why (found the error by trial-and-error not by thinking). My guess is that StructureSetLabel can only be a certain number of digits but my google fu has insofar not given me any conformation or a guide telling my if that is the case and - if yes - how many digits are allowed.

**Comment by grzanka** [comment]
StructureSetLabel is tag (3006,0002), represented as SH (Short string) which is 16 characters maximum (16 chars maximum (see Note in Section 6.2))

Issue #241 might help to prevent such problems in the future.

```
>>> s = "PyTRiP Structures"
>>> len(s)
17
```

I will fix this issue now.

**Comment by grzanka** [comment]
@TokePrintz  fix on the way, see PR #291

**Comment by grzanka** [comment]
To check DICOM tags limitations I am using these two pages:
 - https://www.dabsoft.ch/dicom/6/6/#(0020,1041)
 - ftp://dicom.nema.org/medical/DICOM/2013/output/chtml/part05/sect_6.2.html
